pkg/hubble/metrics/api/api.go

```diff
@@ -126,13 +126,13 @@ func (h Handlers) ProcessFlow(ctx context.Context, flow *pb.Flow) error {
 	return errs
 }
 
-// ProcessPodDeletion queries all handlers for a list of MetricVec and removes
-// metrics directly associated to deleted pod.
-func (h Handlers) ProcessPodDeletion(pod *slim_corev1.Pod) {
+// ProcessCiliumEndpointDeletion queries all handlers for a list of MetricVec and removes
+// metrics directly associated to pod of the deleted cilium endpoint.
+func (h Handlers) ProcessCiliumEndpointDeletion(endpoint *types.CiliumEndpoint) {
```

Comment:
It's been a while since I've looked at the CEP watchers, but would this still work if CiliumEndpointSlice is enabled (i.e. in which case I believe only the operator watches CEPs and agents watch CES resources).

Reply:
Yes, but this is called from `K8sCiliumEndpointsWatcher.endpointDeleted` which is also called from the CES watcher here:

https://github.com/cilium/cilium/blob/8190bf8ddb0f6c48a4d87064fd25da15a365ad1b/pkg/k8s/watchers/cilium_endpoint_slice_subscriber.go#L166